### DISCUSSION TITLE
Any way I could help out w/ thread support?

### DISCUSSION THREAD

**Discussion by vlauffer** [Rapptz/discord.py#7933] [General] [answered]
Using this library to track emoji usage in a discord group (and its awesome, ty so much ), but there's not currently a way to get messages in threads. Is there any way I could help with threads in order to fast-track this feature? I saw that there's thread support in the master branch... is it working as of now or is there still a lot to do in order for it to be functional?

**Comment by LeoCx1000** [comment, marked as answer]
Yes it is working as of ~~now~~ 8 months ago on the master branch.

**Reply by vlauffer** [reply]
oh thats awesome! 

Also for my production environment I'm using heroku to run the python bot, with a requirements.txt file included. Right now in requirements.txt I have discord-py==1.7.3 but is there a way to get the development build by changing the version to like 1.7.4 or something?

**Reply by LeoCx1000** [reply]
You'd put `git+https://github.com/rapptz/discord.py` there instead.
(without specifying a version with ==, just that line I believe.)

**Reply by vlauffer** [reply]
Hey so I think I might have found an issue with threads in the master build. The thread.history function returns messages with empty `content` parameters. It's weird because the `content` parameters for reactions on those messages are showing the correct emojis. Simple example code below would just log empty new lines. 
```
async for message in thread.history(limit=5):
   print(message.content)
```
Is this a known issue? Or am I not calling the function correctly?

**Reply by LostLuma** [reply]
The master branch uses a newer API version in which the message content intent is now required to be enabled if you want to read the actual content of a message.

You will have to enable it in the developer portal and also in your code when starting your client.

**Reply by vlauffer** [reply]
@LostLuma wow this was the easiest fix I've had all day! Ty so much

**Reply by vlauffer** [reply]
Hey I switched to the dev branch and thread tracking is working! Except now the messages from the on_message() function all have empty content parameters. I tried turning on PRESENCE INTENT, SERVER MEMBERS INTENT, and MESSAGE CONTENT INTENT and none of these have fixed the issue. Is there a different intent that I need to check off? Here's a list of my current permissions: 

![Screen Shot 2022-04-27 at 7 43 47 PM](https://user-images.githubusercontent.com/23345792/165648814-1200b61e-3b21-4979-8150-75774d8b62d7.png)

**Reply by LeoCx1000** [reply]
You need the message_content privileged intent explicitly enabled **in code** (by passing the intents kwarg to bot) and in the developer portal

**Reply by vlauffer** [reply]
Yea that was it haha. I had the intents for the client as discord.Intents.default() but when I switched it to discord.Intents.all() it started to work!
